cmake/Modules/HandleSanitizer.cmake

```diff
@@ -88,8 +88,11 @@ if (GALOIS_USE_SANITIZE_COVERAGE)
 endif()
 
 if (GALOIS_USE_SANITIZER MATCHES ".*Undefined.*")
-  set(BLACKLIST_FILE "${CMAKE_SOURCE_DIR}/config/sanitizers/ubsan_blacklist.txt")
-  if (EXISTS "${BLACKLIST_FILE}")
+  message("CURRENT SOURCE DIR: ${CMAKE_CURRENT_SOURCE_DIR}")
+  set(BLACKLIST_CONFIGURE_FILE "${CMAKE_CURRENT_SOURCE_DIR}/config/sanitizers/ubsan_blacklist.txt.in")
+  if (EXISTS "${BLACKLIST_CONFIGURE_FILE}")
+    set(BLACKLIST_FILE "${CMAKE_CURRENT_BINARY_DIR}/config/sanitizers/ubsan_blacklist.txt")
```

Comment:
maybe `${PROJECT_BINARY_DIR}` and `${PROJECT_SOURCE_DIR}` to better reflect the fact these are basically project-wide/global changes rather than scoped to a subdirectory (`CMAKE_CURRENT_XYZ`)?

Reply:
I used `${CMAKE_CURRENT_SOURCE_DIR}` and `${CMAKE_CURRENT_BINARY_DIR}` because some people like to include dependencies as submodules and then connect them using `add_subdirectory`. I haven't checked that this use-case works with Galois, but I'm trying to avoid ruling it out.

Reply:
`${CMAKE_BINARY_DIR}` is certainly bad for someone using `add_subdirectory(external/galois)`

`${PROJECT_BINARY_DIR}` and `${CMAKE_CURRENT_BINARY_DIR}` will refer to the same directory under `add_subdirectory(external/galois)` as the project root is simply the closest directory that contains a `CMakeLists.txt` that has a `project` command.

The project root and the current directory would differ if someone did `add_subdirectory(external/galois/deeper/in/the/tree)` but `HandleSantizer` is only included in `galois/CMakeLists.txt` so this path wouldn't activate.

Another case where they would differ is if someone `include`d or modified their CMake module path to pick up `HandleSanitizer.cmake`. Since we add the path prefix `/config/santizers/`, that's why I thought `${PROJECT_BINARY_DIR}` made more sense otherwise we'd look for the config file based on the where the CMakeLists.txt that included `HandleSanitizer.cmake` was rather than the project root.

Anyway, I'm ok with either `${PROJECT_XYZ}` or `${CMAKE_CURRENT_XYZ}`

Reply:
Got it. For some reason I was thinking that `PROJECT_BINARY_DIR` would just refer to the root build directory. In that case, `PROJECT_SOURCE_DIR` and `PROJECT_BINARY_DIR` are also good here. I'll change it.